runtime/lua/vim/lsp/util.lua

```diff
@@ -868,5 +873,24 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+-- TODO replace with a better implementation.
+function M.json_encode(data)
+  local status, result = pcall(vim.fn.json_encode, data)
+  if status then
+    return result
+  else
+    return nil, result
+  end
+end
+
+function M.json_decode(data)
+  local status, result = pcall(vim.fn.json_decode, data)
+  if status then
+    return result
```

Comment:
if we move these functions to utils, at least return the error message along with status (aka `result`)

Reply:
Thank you for your review.

Is your mean I should return `result, nil` instead of `result`?
If so, I will do it.

Reply:
nevermind, just rebase it's a popular feature :)